tool/tsh/tsh.go

```diff
@@ -783,6 +826,22 @@ func onSSH(cf *CLIConf) {
 		return tc.SSH(cf.Context, cf.RemoteCommand, cf.LocalExec)
 	})
 	if err != nil {
+		if strings.Contains(utils.UserMessageFromError(err), "err-server-is-ambiguous") {
+			allNodes, err := tc.ListAllNodes(cf.Context)
+			if err != nil {
+				utils.FatalError(err)
+			}
+			var nodes []services.Server
+			for _, node := range allNodes {
+				if node.GetHostname() == tc.Host {
+					nodes = append(nodes, node)
+				}
+			}
+			fmt.Fprintf(os.Stderr, "error: ambiguous host could match multiple nodes\n\n")
+			showNodes(nodes, true)
+			fmt.Fprintf(os.Stderr, "Hint: try addressing the node by id (ex: tctl ssh --as-uuid user@node-id)\n")
```

Comment:
`ex: tsh ssh --as-uuid user@node-id`

How about we add `-U` as shorthand of this feature. It's not currently taken by OpenSSH / TSH.